mkosi/distributions/fedora.py

```diff
@@ -50,59 +86,6 @@ def parse_fedora_release(release: str) -> tuple[str, str]:
         return (release, release)
 
 
-@complete_step("Installing Fedora Linux…")
-def install_fedora(state: MkosiState) -> None:
-    release, releasever = parse_fedora_release(state.config.release)
-
-    if state.config.local_mirror:
-        release_url = f"baseurl={state.config.local_mirror}"
-        updates_url = None
-    elif state.config.mirror:
-        baseurl = urllib.parse.urljoin(state.config.mirror, f"releases/{release}/Everything/$basearch/os/")
-        media = urllib.parse.urljoin(baseurl.replace("$basearch", state.config.architecture), "media.repo")
-        if not url_exists(media):
-            baseurl = urllib.parse.urljoin(state.config.mirror, f"development/{release}/Everything/$basearch/os/")
-
-        release_url = f"baseurl={baseurl}"
-        updates_url = f"baseurl={state.config.mirror}/updates/{release}/Everything/$basearch/"
-    else:
-        release_url = f"metalink=https://mirrors.fedoraproject.org/metalink?repo=fedora-{release}&arch=$basearch"
-        updates_url = (
-            "metalink=https://mirrors.fedoraproject.org/metalink?"
-            f"repo=updates-released-f{release}&arch=$basearch"
-        )
-    if release == 'rawhide':
-        # On rawhide, the "updates" repo is the same as the "fedora" repo.
-        # In other versions, the "fedora" repo is frozen at release, and "updates" provides any new packages.
-        updates_url = None
-
-    if releasever in FEDORA_KEYS_MAP:
-        gpgid = f"keys/{FEDORA_KEYS_MAP[releasever]}.txt"
-    else:
-        gpgid = "fedora.gpg"
-
-    gpgpath = Path(f"/etc/pki/rpm-gpg/RPM-GPG-KEY-fedora-{releasever}-{state.config.architecture}")
-    gpgurl = urllib.parse.urljoin("https://getfedora.org/static/", gpgid)
-
-    repos = [Repo("fedora", release_url, gpgpath, gpgurl)]
-    if updates_url is not None:
-        repos += [Repo("updates", updates_url, gpgpath, gpgurl)]
-
-    setup_dnf(state, repos)
-
-    invoke_dnf(state, "install", ["filesystem", *state.config.packages])
-
-    # Fedora defaults to sssd authselect profile, let's override it with the minimal profile if it exists and
-    # extend it with the with-homed feature if we can find it.
-    if state.root.joinpath("usr/share/authselect/default/minimal").exists():
-        run_workspace_command(state, ["authselect", "select", "minimal"])
-
-        features = run_workspace_command(state, ["authselect", "list-features", "minimal"],
-                                         stdout=subprocess.PIPE).stdout.split()
-        if "with-homed" in features:
-            run_workspace_command(state, ["authselect", "enable-feature", "with-homed"])
```

Comment:
I guess dropping this is fine, you only added it a month ago, but I can also see some appeal in leaving it in, because cleaning up the installation of SSSD in a postinst seems slightly more icky to me than starting of with a cleaner, minimal situation. 

Is there maybe a way to cleanly do this in with a skeleton?

Reply:
It's very hard to do this correctly when you consider base images and such. If we do this in install(), it won't always apply since authselect might not always be installed immediately. If we do this in install_packages(), we might be overwriting the postinst script from a base image. This is why I think it's better to leave this to the user to configure.

Also, the debian preset issue we ran into in systemd shows that mkosi mucking around with distro defaults can be very confusing for users. People expect their distro to behave the same as usual regardless of whether it's installed by mkosi or not, which shows that we shouldn't modify defaults unless it's really important.